src/storage/azure/helper.c

```diff
@@ -43,16 +43,34 @@ storageAzureHelper(const unsigned int repoIdx, const bool write, StoragePathExpr
             uriStyle = storageAzureUriStylePath;
     }
 
+    // Ensure the key is valid base64 when key type is shared
+    const StorageAzureKeyType keyType = (StorageAzureKeyType)cfgOptionIdxStrId(cfgOptRepoAzureKeyType, repoIdx);
+    const String *const key = cfgOptionIdxStr(cfgOptRepoAzureKey, repoIdx);
+
+    if (keyType == storageAzureKeyTypeShared)
+    {
+        TRY_BEGIN()
+        {
+            bufNewDecode(encodeBase64, key);
+        }
+        CATCH(FormatError)
+        {
+            THROW_FMT(
+                OptionInvalidValueError, "invalid value for '%s' option: %s\n"
+                "HINT: value must be valid base64 when '%s = " CFGOPTVAL_REPO_AZURE_KEY_TYPE_SHARED_Z "'.",
+                cfgOptionIdxName(cfgOptRepoAzureKey, repoIdx), errorMessage(), cfgOptionIdxName(cfgOptRepoAzureKeyType, repoIdx));
+        }
+        TRY_END();
+    }
+
```

Comment:
I don't think this is the best place for this error message, because we don't have information about which repo is being configured. It seems best to do this in the helper. See 2787d9b1a1a.

Reply:
Oops, this is the new code, but you get the idea.

Reply:
I do. Added a note to self to think in terms of scaled/non simple operations (i.e. up to 256 repos, etc ) and does this change appropriately provide for that.